src/hooks/useMixpanel.ts

```diff
@@ -0,0 +1,24 @@
+// eslint-disable-next-line no-restricted-imports
+import mixpanel, { Dict, RequestOptions, Callback } from "mixpanel-browser";
+import { useMemo } from "react";
+
+// https://stackoverflow.com/questions/32203420/check-if-mixpanel-library-has-been-loaded
+const isLoaded = () => mixpanel.hasOwnProperty("get_distinct_id");
+
+const noopTrack: (
+  event_name: string,
+  properties?: Dict,
+  optionsOrCallback?: RequestOptions | Callback,
+  callback?: Callback
+) => void = () => {};
+
+export const useMixpanel = () => {
+  const mixpanelMemo = useMemo(
+    () => ({
+      track: isLoaded() ? mixpanel.track.bind(mixpanel) : noopTrack,
```

Comment:
Because this is binding once the first time this hook is called, it will never break memo to use the correct callback.

I believe what we would want to do here is something like (untested):
```suggestion
      track: () => isLoaded() ? mixpanel.track : noopTrack,
```

Reply:
The `bind` is needed because `mixpanel.track` has references to `this` in the code; it crashes without the `bind`.

Reply:
And the suggestion changes the function signature I think such that you'd need to wrap the `track` call. My goal was to make this safety hook as pass-through as possible, so you don't need to think about how it works. Which follows the precedent of the `useDispatch` and `useSelector` we already have.

Reply:
We need the wrapper so that it is evaluated on every call, so the suggestion may not be perfectly accurate as is, but we need to do it and wrap our if/etc logic. Otherwise it will be evaluated once at app load and never again.

Reply:
OK, yep, I see. Thanks for the explanation. Looking at this.